Mathlib/ModelTheory/Syntax.lean

```diff
@@ -366,11 +366,21 @@ protected def ex (φ : L.BoundedFormula α (n + 1)) : L.BoundedFormula α n :=
 instance : Top (L.BoundedFormula α n) :=
   ⟨BoundedFormula.not ⊥⟩
 
+/-- The conjunction of two bounded formulas is also a bounded formula. -/
+@[match_pattern]
+protected def inf (φ ψ : L.BoundedFormula α n) : L.BoundedFormula α n :=
+  (φ.imp ψ.not).not
+
 instance : Inf (L.BoundedFormula α n) :=
-  ⟨fun f g => (f.imp g.not).not⟩
+  ⟨fun f g => f.inf g⟩
+
+/-- The disjunction of two bounded formulas is also a bounded formula. -/
+@[match_pattern]
+protected def sup (φ ψ : L.BoundedFormula α n) : L.BoundedFormula α n :=
+  φ.not.imp ψ
 
 instance : Sup (L.BoundedFormula α n) :=
-  ⟨fun f g => f.not.imp g⟩
+  ⟨fun f g => f.sup g⟩
```

Comment:
What's the point of this change?

Reply:
I defined separate `sup` and `inf` functions with a `match_pattern` tag because to be able to use pattern matching in the definition of `simpleNot`. Do you know how to make the expressions `φ ⊔ ψ` and `φ ⊓ ψ` matchable?

Reply:
`attribute [match_pattern] Sup.sup Inf.inf`. You should add this in `Mathlib.Order.Notation` (and it can be a separate PR)